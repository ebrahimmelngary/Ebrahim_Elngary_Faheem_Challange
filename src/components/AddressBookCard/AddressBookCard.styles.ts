import { StyleSheet } from 'react-native';
import { COLORS } from '../../common/colors';

export default StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    padding: 10,
    marginBottom: 10,
    backgroundColor: COLORS.lightGray,
    marginHorizontal: 16,
    borderRadius: 8,
  },
  picture: {
    width: 30,
    height: 30,
    borderRadius: 30,
  },
  details: {
    paddingHorizontal: 10,
  },
});
