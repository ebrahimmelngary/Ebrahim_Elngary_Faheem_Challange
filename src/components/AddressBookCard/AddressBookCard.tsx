import React from 'react';
import { Text, View } from 'react-native';
import FastImage from 'react-native-fast-image';

import { Result } from '../../types';
import styles from './AddressBookCard.styles';

const AddressBookCard = ({ item }: { item: Result }) => {
  return (
    <View style={styles.container}>
      <FastImage
        source={{ uri: item?.picture?.thumbnail }}
        resizeMode={FastImage.resizeMode.contain}
        style={styles.picture}
      />
      <View style={styles.details}>
        <Text>{`${item?.name?.title} ${item?.name?.first} ${item?.name?.last}`}</Text>
        <Text>{item?.email}</Text>
      </View>
    </View>
  );
};

export default AddressBookCard;
