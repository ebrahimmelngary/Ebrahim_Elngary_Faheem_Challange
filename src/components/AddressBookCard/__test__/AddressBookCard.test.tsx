import React from 'react';
import { render, cleanup } from '@testing-library/react-native';
import AddressBookCard from '../AddressBookCard';
import { Result } from '../../../types';

describe('AddressBookCard', () => {
  let props: JSX.IntrinsicAttributes & { item: Result };

  afterEach(cleanup);
  it('should match snapshot', () => {
    const { toJSON } = render(<AddressBookCard {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
