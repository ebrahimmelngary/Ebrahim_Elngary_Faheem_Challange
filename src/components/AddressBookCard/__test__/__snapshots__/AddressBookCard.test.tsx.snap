// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressBookCard should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#bfbfbf",
      "borderRadius": 8,
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 10,
      "marginHorizontal": 16,
      "padding": 10,
    }
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderRadius": 30,
          "height": 30,
          "width": 30,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="contain"
      source={
        {
          "uri": undefined,
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      {
        "paddingHorizontal": 10,
      }
    }
  >
    <Text>
      undefined undefined undefined
    </Text>
    <Text />
  </View>
</View>
`;
